### DISCUSSION TITLE
Method LaravelZero\Framework\Application::originalFileName does not exist.

### DISCUSSION THREAD

**Discussion by ousid** [laravel-zero/laravel-zero#486] [Q&A] [answered]
Trying to use `spatie/medialibrary-laravel` inside laravel zero, and fetching images, from a url, but I get the following error: `Method LaravelZero\Framework\Application::originalFileName does not exist.`

### Steps To Produce
- install media library package
- Fetch image url
- Store the image into a collection (s3)
- Run the command

### Code Sample
```php
collect($images)->each(function ($imageUrl) use ($productId) {
  $product = Product::find($productId);
  $product->addMediaFromUrl($imageUrl)
     ->toMediaCollection('images', 's3');
});
```

Did anyone ran into a similar issue, and found a solution around?

Thanks in advance!

**Comment by owenvoke** [comment, marked as answer]
Have you added `Spatie\MediaLibrary\MediaLibraryServiceProvider::class` to your service providers in `config/app.php`? 🤔

My guess is that the config hasn't been loaded because the service provider isn't loaded, and that means that `media-library.file_namer` is `null`, so [this line](https://github.com/spatie/laravel-medialibrary/blob/main/src/MediaCollections/FileAdder.php#L282) is probably just resolving `app()` as the actual application.

**Reply by ousid** [reply]
You're totally right, I forgot about adding the service provider, in the `config/app.php`, since I'm a Laravel guy, I thought there's an __auto discovery__ in Laravel zero 😆 

Thank you for the help, @owenvoke!

**Reply by owenvoke** [reply]
Yeah, unfortunately not. 😬 I have wanted to add auto-discovery, however there were some issues I remember.

**Reply by ousid** [reply]
I see! Thanks for all the hard work, and for this awesome project @owenvoke!
